### PR TITLE
Updated strict concurrency and Privacy Manifests

### PR DESCRIPTION
**Pull request by gazreese**
This PR supersedes and includes https://github.com/Flagsmith/flagsmith-ios-client/pull/48.

This PR updates the above by also updating the Cocoapods functionality to match the Swift Package Manager changes above: https://github.com/Flagsmith/flagsmith-ios-client/pull/50/commits/6b5dc80a159e04252c1265f1f7f1696f9ca94e1f

Due to upcoming https://developer.apple.com/support/third-party-SDK-requirements/ we've updated the SDK to include a Privacy Manifest.

The format and purpose of these is described here: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files

The crux of what I've gone for on here is:

UserDefaults: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api#4278401

This is needed to store persistent state info for the SDK. At the moment this is just used to store analytics events in the case they can't be sent immediately. This could possibly be removed in the future if you're (a) not that fussed and we (b) post analytics when the app goes to the background to avoid data loss.

Other Diagnostic Data:

image

from here: https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests#4144131

The release is set up to be version 3.6.0 in the Podspec.

I've tested this with the unit tests and also creating local packages with Swift Package Manager and Cocoapods to see if the Privacy Manifest is included, which it does appear to be. When running the new SDK in the example app I didn't see any additional privacy dialogs appear.

I've also checked my changes against other popular iOS SDKs and can confirm they're consistent.

Manifest files are typically checked by Apple automatically when submitting to the App Store, so that's going to be the real test for this one.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [x] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

**Comment by dabeeeenster** [comment]
Yep the Privacy Manifest stuff LGTM here.

File: .github/workflows/pull-request.yml [hunk 1]
```diff
@@ -8,10 +8,10 @@ on:
 
 jobs:
     macos-build:
-        runs-on: macos-latest
+        runs-on: macOS-13
```
Comment (dabeeeenster) [inline .github/workflows/pull-request.yml:11]: Whats the reason for this change? `latest` would seem like a better choice?

Comment (gazreese) [reply .github/workflows/pull-request.yml:11]: It does @dabeeeenster, this seems to have come with the original PR. I'll switch it back and check that everything builds...

Comment (gazreese) [reply .github/workflows/pull-request.yml:27]: Hi @dabeeeenster if you check out this file again I've added a load of comments that should be self-explanatory. Turns out latest is like 3/4 years old, so specifying the version actually gets us newer xCode / MacOS.

I've added 13 and 14 to cover us for now. 12 (latest) didn't have the build tools we need.

Comment (dabeeeenster) [reply .github/workflows/pull-request.yml:27]: Huh weird. OK yep hard coding sounds like the only path.

**Review by matthewelwell — COMMENTED** [review]
I've added a couple of comments on the privacy manifest, but I'm a little confused why this PR seems to include t2o totally distinct things. I'm not sure why we merged the strict concurrency PR into a branch to then add the manifest to as well. 

I guess my main concern is that the description of the PR references the changes to the privacy manifest but mentions nothing about any testing that has been carried out on what seem like fairly hefty changes on the actual SDK.

**Comment by matthewelwell** [comment]
Superseded by #55

**Pull request closed without merging**